compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -12810,6 +12810,149 @@ TR::Node *su2dSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier * s)
    return node;
    }
 
+
+/**
+ * \brief
+ * Remove add/subtract operations under comparison nodes to save an
+ * addition or subtraction instruction during codegen. This simplification
+ * applies to comparisons of byte, short, int, and long values.
+ *
+ * \details
+ *
+ * \verbatim
+ * As an example, assuming that the following is the original expression
+ *        x + oldConst1 > oldConst2
+ *
+ * This expression can be simplified to
+ *        x > newConst
+ *        where newConst = oldConst2 - oldConst1
+ *
+ * \endverbatim
+ *
+ *  And the newConst calculation should not produce overflow or underflow.
+ *
+ * \param node          the comparison node being simpilfied
+ * \param s             the simplifier object
+*/
+TR::Node* removeArithmeticsUnderIntegralCompare(TR::Node* node,
+                                                TR::Simplifier * s)
+   {
+   if (s->comp()->getOption(TR_DisableIntegerCompareSimplification))
+      return node;
+
+   TR::Node* opNode      = node->getFirstChild();
+   TR::Node* secondChild = node->getSecondChild();
+   TR::ILOpCodes op      = opNode->getOpCodeValue();
+   bool isInt            = !opNode->getOpCode().isLong();
+
+   // Get recognized operation types. Unsigned IL OpCodes are deprecated and not supported here.
+   bool isAddOp = (op == TR::iadd || op == TR::ladd || op == TR::sadd || op == TR::badd);
+   bool isSubOp = (op == TR::isub || op == TR::lsub || op == TR::ssub || op == TR::bsub);
+
+   if ((isAddOp || isSubOp) &&
```

Comment:
this is easier to read if you put the && at the start of each line rather than the end of the line before. The indent seems very large here in the browser - can you line up on the ( of (isAddOp || .. or something else reasonable in that area - indent is 3 levels at a time in this code :)